src/pymor/reductors/mt.py

```diff
@@ -0,0 +1,197 @@
+from pymor.operators.constructions import IdentityOperator
+from pymor.operators.numpy import NumpyMatrixOperator
+from pymor.parameters.base import Mu
+from pymor.reductors.basic import LTIPGReductor
+
+
+class MTReductor(BasicObject):
+    """Modal Truncation reductor.
+
+    See Section 9.2 in [A05]_.
+
+    Parameters
+    ----------
+    fom
+        The full-order |LTIModel| to reduce.
+    mu
+        |Parameter values|.
+    """
+
+    def __init__(self, fom, mu=None):
+        assert isinstance(fom, LTIModel)
+        if not isinstance(mu, Mu):
+            mu = fom.parameters.parse(mu)
+        assert fom.parameters.assert_compatible(mu)
+        self.fom = fom
+        self.mu = mu
+        self.V = None
+        self.W = None
+        self._pg_reductor = None
+
+    def reduce(self, r=None, decomposition='samdp', projection='orth',
+               symmetric=False, which='NR', method_options=None, allow_complex_rom=False):
+        """Modal Truncation.
+
+        Parameters
+        ----------
+        r
+            Order of the reduced model.
+        decomposition
+            Algorithm used for the decomposition:
+
+            - `'eig'`: scipy.linalg.eig algorithm
+            - `'samdp'`: find dominant poles using
+              :meth:`~pymor.algorithms.samdp.samdp` algorithm
+        projection
+            Projection method used:
+
+            - `'orth'`: projection matrices are orthogonalized with
+              respect to the Euclidean inner product
+            - `'biorth'`: projection matrices are biorthogolized with
+              respect to the E product
+        symmetric
+            If `True`, assume A is symmetric and E is symmetric positive
+            definite.
+        which
+            A string specifying which `r` eigenvalues and eigenvectors to
+            compute when using the eig decomposition.
+            Possible values are:
+
+            - `'SM'`: select eigenvalues with smallest magnitude
+            - `'LR'`: select eigenvalues with largest real part
+            - `'NR'`: select eigenvalues with largest norm(residual) / abs(Re(pole))
+            - `'NS'`: select eigenvalues with largest norm(residual) / abs(pole)
+            - `'NM'`: select eigenvalues with largest norm(residual)
+        method_options
+            Optional dict with more options for the samdp algorithm.
+        allow_complex_rom
+            If `True`, the reduced model is complex when the poles of the reduced
+            model are not closed under complex conjugation.
+
+        Returns
+        -------
+        rom
+            Reduced-order model.
+        """
+        assert 0 < r < self.fom.order
+        assert projection in ('orth', 'biorth')
+        assert decomposition in ('eig', 'samdp')
+        assert which in ('LR', 'SM', 'NR', 'NS', 'NM')
+        assert method_options is None or isinstance(method_options, dict)
+        if not method_options:
+            method_options = {'which': 'LR'}
```

Comment:
I think the default `which` argument for `samdp` should depend on the `which` argument of the `MTReductor.reduce` method. E.g. if `which == 'NS'` then the default option should be `method_options = {'which': 'LS'}` instead of `method_options = {'which': 'LR'}` in order to ensure that `samdp` returns exactly the poles that we are looking for. Similarly `method_options` should be adjusted if  `which == 'NM'` or `which == 'NR'`.

For the 'LR' and 'SM' cases I would use `method_options = {'which': 'LM'}`.

Reply:
Thanks for your review. 
As stated in the docstring my intention was that the which argument is only used when the eig decomposition is used otherwise the sorting for samdp should be configured with the method_options dict. But you may be right that it seems more intuitive to use the which argument for both decomposition algorithms.

Reply:
I had been thinking about it. It would indeed be nice if `which` can be used for both algorithms, especially once support for `eigs` is added, but it is a bit unfortunate that the option strings do not match. @lbalicki What do you think about changing the `which` options in `samdp`?

Then, if `decomposition == 'samdp'`, I would propose raising a `ValueError` if `which in ('LR', 'SM')` (and similarly for the potential future `eigs` support).

Reply:
I did the renaming in  #1247, this way the `which` argument of `MTReductor.reduce` can be passed to `samdp`. I think it will be more straightforward for a user if `which` is used for all the methods that compute the poles. I guess further arguments for `samdp` can still be passed through the `method_options`.